odxtools/diaglayer.py

```diff
@@ -636,9 +643,73 @@ def get_comparam(
 
         return cps[0]
 
-    def get_can_receive_id(self, protocol_name: Optional[str] = None) -> Optional[int]:
+    def get_can_mtu(self, protocol: Optional[Union[str, "DiagLayer"]] = None) -> Optional[int]:
+        """Return the maximum size of CAN frames that can be
+        transmitted in bytes.
+
+        For classic CAN busses, this is basically always 8. CAN-FD can
+        send up to 64 bytes per frame.
+        """
+        com_param = self.get_comparam("CP_CANFDTxMaxDataLength", protocol=protocol)
+        if com_param is None:
+            rx_id = self.get_can_receive_id(protocol=protocol)
+            if rx_id is not None:
+                # bus is CAN. We also use 8 bytes for CAN-FD if this
+                # parameter was not specified.
+                return 8
+
+            # bus is not CAN
+            return None
+
+        val = com_param.value
+        if not isinstance(val, str):
+            return None
+
+        m = re.search("TX_DL *= *([0-9]*)", val)
+        if m:
+            return int(m.group(1))
+
+        # unexpected format of parameter value
+        return 8
+
+    def use_can_fd(self, protocol: Optional[Union[str, "DiagLayer"]] = None) -> bool:
```

Comment:
Maybe return None if inconclusive?
```suggestion
    def use_can_fd(self, protocol: Optional[Union[str, "DiagLayer"]] = None) -> Optional[bool]:
```

Reply:
what does `None` mean in this context? "I don't know"? I'd expect to go for classical CAN if CAN FD is not specified explicitly. The thing I'm not 100% sure is if other manufacturers specify using CAN FD the same way as Mercedes does. (I assume that they are, mainly because the number of available commercial tools is rather limited AFAICS. If there are other ways to specify CAN FD in the wild: patches welcome ;))